src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -1859,29 +1865,36 @@ type internal TransparentCompiler
             }
         )
 
-    let ComputeAssemblyData (projectSnapshot: ProjectSnapshot) fileName =
+    let ComputeAssemblyData (projectSnapshot: ProjectSnapshot) _fileName =
         caches.AssemblyData.Get(
             projectSnapshot.SignatureKey,
             async {
+                use _ =
+                    Activity.start
+                        "ComputeAssemblyData"
+                        [|
+                            Activity.Tags.project, projectSnapshot.ProjectFileName |> Path.GetFileName |> (!!)
+                        |]
+
                 use! _holder = Cancellable.UseToken()
 
                 try
 
-                    let availableOnDiskModifiedTime =
-                        if FileSystem.FileExistsShim fileName then
-                            Some <| FileSystem.GetLastWriteTimeShim fileName
-                        else
-                            None
+                    //let availableOnDiskModifiedTime =
+                    //    if FileSystem.FileExistsShim fileName then
+                    //        Some <| FileSystem.GetLastWriteTimeShim fileName
+                    //    else
+                    //        None
 
                     // TODO: This kinda works, but the problem is that in order to switch a project to "in-memory" mode
                     //  - some file needs to be edited (this triggers a re-check, but LastModifiedTimeOnDisk won't change)
                     //  - saved (this will not trigger anything)
                     //  - and then another change has to be made (to any file buffer) - so that recheck is triggered and we get here again
                     // Until that sequence happens the project will be used from disk (if available).
                     // To get around it we probably need to detect changes made in the editor and record a timestamp for them.
-                    let shouldUseOnDisk =
-                        availableOnDiskModifiedTime
-                        |> Option.exists (fun t -> t >= projectSnapshot.GetLastModifiedTimeOnDisk())
+                    let shouldUseOnDisk = false
+                    //availableOnDiskModifiedTime
+                    //|> Option.exists (fun t -> t >= projectSnapshot.GetLastModifiedTimeOnDisk())
```

Comment:
Could this be removed and rephrased with something more actionable?
I tried reading it multiples times and I did not get an understanding of how big of an issue that is, what has to change or if it is fine to just not do that at all.